### ISSUE TITLE
llava cannot produce diverse output sentence in any SamplingConfig

### ISSUE THREAD

**Issue by jinluyang** [NVIDIA/TensorRT-LLM#1407]
### System Info

- A10 GPU
- TensorRTLLM == v0.9.0
- TensorRT == 9.2.0.post12.dev5

### Who can help?

@byshiue 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

git clone https://huggingface.co/llava-hf/llava-1.5-13b-hf 
modify examples/multimodal/run.py and add sampling config in its generate function:

```
@@ -189,7 +189,9 @@ class MultiModalModel:
                 top_k=self.args.top_k,
                 num_beams=self.args.num_beams,
                 output_sequence_lengths=False,
-                return_dict=False)
+                return_dict=False,
+                temperature=1.0,
+                do_sample=do_sample)
         else:
             if args.nougat:

``` 
although the do_sample parameter do not work , I also added top_k top_p params to be 10 and 0.5

then follow the README in the example, run a prompt several times and it provides the exactly same result.

### Expected behavior

should produce different sentence when temperature is high

### actual behavior

always produce the same sentence

### additional notes

I printed the SamplingConfig right before the call of each self.session.decode in model_runner.py and it is :

SamplingConfig(end_id=2, pad_id=2, max_new_tokens=128, num_beams=1, max_attention_window_size=None, sink_token_length=None, output_sequence_lengths=False, return_dict=False, stop_words_list=None, bad_words_list=None, temperature=1.0, top_k=10, top_p=0.5, top_p_decay=None, top_p_min=None, top_p_reset_ids=None, length_penalty=1.0, repetition_penalty=1.0, min_length=1, presence_penalty=0.0, frequency_penalty=0.0, use_beam_hyps=True, beam_search_diversity_rate=0.0, random_seed=1, output_cum_log_probs=False, output_log_probs=False)

**Comment by byshiue** [comment]
If you run the `run.py` several times, you should get same results because you use same `random_seed`. If you want to get different results, you need to set `random_seed` in `SamplingConfig`.

**Comment by jinluyang** [comment]
> If you run the `run.py` several times, you should get same results because you use same `random_seed`. If you want to get different results, you need to set `random_seed` in `SamplingConfig`.

Thanks, now I set random_seed=time.time(), so the result now varies from time to time. So now I understand temperature only control the confidence of the model about its logits but we must change random_seed to make it generate a different output. The `random_seed` parameter is controling the same thing compared to the `do_sample` parameter in the huggingface `generate`. Am I right sir?

**Comment by byshiue** [comment]
Huggingface should also have random seed, it might be controlled by torch random seed directly. 
`do_sample = False` should be more close to `top_k=1`, which uses greedy search.
